src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -1878,15 +1880,15 @@ def test_ps_s3_lifecycle_on_master():
 
     # start lifecycle processing
     admin(['lc', 'process'], get_config_cluster())
-    print('wait for 5sec for the messages...')
-    time.sleep(5)
+    print('wait for 20s for the lifecycle...')
+    time.sleep(20)
 
-    # check http receiver does not have messages
-    keys = list(bucket.list())
-    print('total number of objects: ' + str(len(keys)))
+    no_keys = list(bucket.list())
+    wait_for_queue_to_drain(topic_name)
+    assert_equal(len(no_keys), 0)
     event_keys = []
     events = http_server.get_and_reset_events()
-    assert_equal(number_of_objects * 2, len(events))
+    assert number_of_objects * 2 <= len(events)
```

Comment:
why change this check to look for  `<= 2* events`
coz the persistent notifications are not delivering all the events ?
since `wait_for_queue_to_drain` is already verified before, then all the events should have been delivered and we should receive `number_of_objects * 2` events

Reply:
persistent notifications may deliver the same event more than once - the delivery guaranteee for persistent notifications is "at least once".
since we have 2 events per object, we should recieve at least twice the number objects